src/tool/xmeta/cpp/ast_to_st_listener.cpp

```diff
@@ -4,52 +4,137 @@
 #include "ast_to_st_listener.h"
 #include "xmeta_idl_reader.h"
 #include "models/xmeta_models.h"
+#include "XlangParser.h"
 
 using namespace xlang::xmeta;
 
-enum_semantics str_to_enum_semantics(std::string const& val)
+namespace
 {
-    if (val == "Int8")
-    {
-        return enum_semantics::Int8;
-    }
-    else if (val == "UInt8")
+    static const std::map<std::string, enum_semantics> str_to_enum_type_map = {
+        { "Int8", enum_semantics::Int8 },
+        { "Int16", enum_semantics::Int16 },
+        { "Int32", enum_semantics::Int32 },
+        { "Int64", enum_semantics::Int64 },
+        { "UInt8", enum_semantics::UInt8 },
+        { "UInt16", enum_semantics::UInt16 },
+        { "UInt32", enum_semantics::UInt32 },
+        { "UInt64", enum_semantics::UInt64 },
+    };
+
+    enum_semantics str_to_enum_semantics(std::string const& val)
     {
-        return enum_semantics::UInt8;
+        auto const iter = str_to_enum_type_map.find(val);
+        if (iter == str_to_enum_type_map.end())
+        {
+            throw std::invalid_argument("Can't convert " + val + "to simple type");
```

Comment:
This runs the risk of diverging from the rest of the xlang code base. Use `xlang::throw_invalid("Can't convert ", val, " to simple type");`